From 6fae2dc7f45d14b5341e9929b5c5fef60596ed4f Mon Sep 17 00:00:00 2001
From: Simon Casey <simon501098c@gmail.com>
Date: Wed, 2 Feb 2022 20:25:38 +0100
Subject: [PATCH] Update
 903-arm64-mm-export-__dma_inv_area-and-__dma_clean_area.patch for kernel 5.15

---
 arch/arm64/include/asm/cacheflush.h | 4 ++--
 arch/arm64/mm/cache.S               | 4 ++--
 arch/arm64/mm/flush.c               | 2 ++
 3 files changed, 6 insertions(+), 4 deletions(-)

--- a/arch/arm64/include/asm/cacheflush.h
+++ b/arch/arm64/include/asm/cacheflush.h
@@ -110,8 +110,8 @@ static inline void flush_icache_range(un
 extern void __dma_map_area(const void *, size_t, int);
 extern void __dma_unmap_area(const void *, size_t, int);
 extern void __dma_flush_area(const void *, size_t);
-extern void __dma_inv_area(const void *start, const void *end);
-extern void __dma_clean_area(const void *start, const void *end);
+extern void __dma_inv_area(const void *, const void *);
+extern void __dma_clean_area(const void *, const void *);
 
 #define dmac_flush_range(start, end) \
 	__dma_flush_area(start, (void *)(end) - (void *)(start))
--- a/arch/arm64/mm/cache.S
+++ b/arch/arm64/mm/cache.S
@@ -140,7 +140,7 @@ SYM_FUNC_END(dcache_clean_pou)
  *	- start   - kernel start address of region
  *	- end     - kernel end address of region
  */
-SYM_FUNC_START_LOCAL(__dma_inv_area)
+SYM_FUNC_START(__dma_inv_area)
 SYM_FUNC_START_PI(dcache_inval_poc)
 	/* FALLTHROUGH */
 
@@ -178,7 +178,7 @@ SYM_FUNC_END(__dma_inv_area)
  *	- start   - virtual start address of region
  *	- end     - virtual end address of region
  */
-SYM_FUNC_START_LOCAL(__dma_clean_area)
+SYM_FUNC_START(__dma_clean_area)
 SYM_FUNC_START_PI(dcache_clean_poc)
 	/* FALLTHROUGH */
 
--- a/arch/arm64/mm/flush.c
+++ b/arch/arm64/mm/flush.c
@@ -77,6 +77,8 @@ EXPORT_SYMBOL(flush_dcache_page);
  * Additional functions defined in assembly.
  */
 EXPORT_SYMBOL(caches_clean_inval_pou);
+EXPORT_SYMBOL(__dma_inv_area);
+EXPORT_SYMBOL(__dma_clean_area);
 
 #ifdef CONFIG_ARCH_HAS_PMEM_API
 void arch_wb_cache_pmem(void *addr, size_t size)
